pkg/fieldpath/fieldpath.go

```diff
@@ -18,26 +18,36 @@ package fieldpath
 
 import (
 	"fmt"
+	"sort"
+	"strconv"
 	"strings"
 
 	"k8s.io/apimachinery/pkg/api/meta"
-	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation"
 )
 
 // FormatMap formats map[string]string to a string.
 func FormatMap(m map[string]string) (fmtStr string) {
 	// output with keys in sorted order to provide stable output
-	keys := sets.NewString()
-	for key := range m {
-		keys.Insert(key)
+	keys := make([]string, 0, len(m))
+	var grow int
+	for k, v := range m {
+		keys = append(keys, k)
+		// why add 3: (for =, " and ")
+		grow += len(k) + len(v) + 3
```

Comment:
actually, I guess this should add 4 to account for the newline as well

Reply:
Fixed

It does miss the line break